locator/abstract_replication_strategy.cc

```diff
@@ -155,13 +170,11 @@ effective_replication_map::do_get_ranges(noncopyable_function<bool(inet_address_
 
 dht::token_range_vector
 effective_replication_map::get_ranges(inet_address ep) const {
-    return do_get_ranges([ep] (inet_address_vector_replica_set eps) {
-        for (auto a : eps) {
-            if (a == ep) {
-                return true;
-            }
+    return do_get_ranges([ep] (bool& add_range, const inet_address& e) {
+        if ((add_range = (e == ep))) {
+            return stop_iteration::yes;
```

Comment:
I'm really confused now. It **looks** to me as this stops the iteration - and stops adding things to the vector - as soon as we find one range. Isn't this incorrect? What am I missing?

Do we have a unit test for this function? Or a higher-level test that we believe exercises this function well?

Reply:
Your understanding is correct.
Note that previously we'd also `return true` and stop the iteration on first match:
```
        for (auto a : eps) {
            if (a == ep) {
                return true;
            }
```

This is tested pretty much by any topology change test.

Reply:
I just realized that there are **two** iterations happening here: There is an iteration over token ranges, and for each token range there is an iteration over the different replicas of this token range. The "stop_iteration" we're seeing here stops the inner iteration, not the outer iteration.

This confused me completely in my review.

I'm not sure what can be done to make this clearer, though.

I'll review again with my new understanding.